### DISCUSSION TITLE
[Modding] Support for NPC merchants is really bad

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#13242] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The way NPC merchant stores are generated is really cumbersome for modding. 

Where items are sold is defined directly inside the item itself, which means if you want an item to be sold in a custom merchant NPC, you need to override each individual item you want to be sold there which causes awful amounts of incompatibility.

Modders resort to vending machines to circumvent the problem entirely.

![image](https://github.com/Regalis11/Barotrauma/assets/73229309/5e57e416-7bcf-4713-bdb0-e726abea94e3)
![image](https://github.com/Regalis11/Barotrauma/assets/73229309/647e2c19-a711-4a9b-a8b3-a862fec9b6b8)

**Suggestion:**

Add support for a "merchantlist" in NPCs, in which you can list individual items, so you don't have to override items themselves. It's better to override 1 NPC than 50 items

![image](https://github.com/Regalis11/Barotrauma/assets/73229309/296f4902-6a81-4439-ad0f-eba6fbbc6c6f)
![image](https://github.com/Regalis11/Barotrauma/assets/73229309/8a294267-1d09-42d2-bd94-826f69a91cba)

### Version

v1.2.7.0 (Winter Update hotfix)

**Comment by ghost** [comment]
> better to override 1 NPC than 50 items

Not really, because 1 NPC can only be overridden by one mod. Say you have 3 mods that want to add or adjust purchasable items in the same merchant, only one mod can have their items in the shop. This is a compatibility **nightmare**

> Modders resort to vending machines to circumvent the problem entirely.

Which mods do this? Could you give some examples?

Also, are vending machines not the same? They're fabricators that use money, so do you not have to define the recipe in the item itself anyway? Why not just add or adjust the Price element if you're messing with / adding the item?

**Reply by ubertpendragon** [reply]
It's already a compatibility nightmare, it's easier to make a compatiblity patch for a few NPCs than all items in the game

DE uses vending machines, but vending machines arent affected by reputation and talent cost modifiers

**Reply by ghost** [reply]
I had a peek at DE's food items, and it seems yes, they do use vending machines as fabricators. I still don't understand how the existence of vending machines changes anything, since it doesn't circumvent the issue.

(Pomegrenade chips from DE:)
```xml
<Fabricate suitablefabricators="vendingmachinefood,cafeteria" requiredtime="3" requiredmoney="19" fabricationlimitmin="0" fabricationlimitmax="10" />fabricationlimitmax="10" />
```

**Comment by Crazyjr20** [comment]
Course, it's never as simple as it sounds. But having the ability to write a new item set sorta deal to add items to npc vendors would be nice. Rather than overriding the npc/item itself. The idea being that you create an item set with a unique identifier, it looks for npcs ids you specify, adds the list of items below with the price you set.
mockup example:
< MerchantList identifier="op_newitems" targetnpc="medicalmerchant">
<Item="opbandage" baseprice="500" minamount="6" maxamount="10"/>
<Item="opmorphine" baseprice="750" minamount="4" maxamount="8"/>
< /MerchantList>

Baseprice might not even be needed, but could be something like newprice instead to let people have more/less expensive merchants.
